File: lab-report-02/insertion-sort.c
//Fatima
//Batch: E-76th
//Roll:19

//insertion sort ascending or descending order
#include<stdio.h>
#include<stdlib.h>
int main(){
    int list[50], N, choice, i, j, temp;

    //menu section
    while(1){
    printf("           Menu             \n");
    printf("1.choose 1 for ascending order sorting:\n");
    printf("2.choose 1 for descending order sorting:\n");
    printf("3.choose 0 for exit:\n");
    printf("choose your choice:\n");
    scanf("%d",&choice);

    if(choice == 0)
    exit(0);

    //input section
    printf("Enter how many numbers you want to sort:");
    scanf("%d", &N);
    printf("\n");

    for(i = 1; i <= N; i++){
    printf("Enter your %d number:",i);
    scanf("%d", &list[i]);
    }

    //descending order
    for(i = 1; i>=N; i--){
    printf("Enter your %d number:",i);
    scanf("%d", &list[i]);
    }
    //insertion sort for ascending order sorting
    if(choice == 1){
        for(i = 2; i<=N-1; i++){
            temp = list[i];
            j = i - 1;
            while(j >= 1 && list [j] > temp){
                list [j+1] = list[j];
                j--;
            }
            list[j+1] = temp;
        }
    }
    else if(choice == 2){
            for(i = 2; i<=N; i++){
            temp = list[i];
            j = i - 1;
            while(j >= 1 && list [j] < temp){
                list [j+1] = list[j];
                j--;
            }
            list[j+1] = temp;
        }
    }

    //output section
    printf("After sorting the list: \n");
    for(i = 1; i <= N; i++){
        printf("%d",list[i]);
    }
    printf("\n");

    }
  return 0;
}

File: linear-bubble-selection/InsertionSort.cs
using System;
using System.Linq;
using static System.Console;

namespace Lab_Task_02
{
    public class InsertionSort
    {
        private static int[] Ascending(int length, int[] numbers)
        {
            int i, j,temp;
            for (i = 1; i < length; i++)
            {
                temp = numbers[i];
                j = i-1;
                while (j>=0 && numbers[j]>temp)
                {
                    numbers[j+1]=numbers[j];
                    j--;
                }
                numbers[j+1] =temp;
            }
            return numbers;
        }
        private static int[] Descending(int length, int[] numbers)
        {
            int i, j, temp;
            for (i = 0; i < length; i++)
            {
                temp = numbers[i];
                j = i-1;
                while (j>=0 && numbers[j]<temp)
                {
                    numbers[j+1] = numbers[j];
                    j--;
                }
                numbers[j+1] = temp;
            }
            return numbers;
        }

        public static void Sort()
        {
            int length = 0, sortType, i;
            Write("How many number you want to sort? ");
            length = int.Parse(ReadLine());
            Write("Enter the numbers: ");
            int[] numbers = ReadLine().Split(" ").Select(num=>int.Parse(num)).ToArray()[0..length];
            WriteLine("............ Menu .............");
            WriteLine("Press 1 for ascending order sorting.");
            WriteLine("Press 2 for descending order sorting.");
            Write("Enter your choice: ");
            sortType = int.Parse(ReadLine());
            if (sortType == 1)
            {
                WriteLine("You have chosen ascending order sorting.");
                numbers =Ascending (length, numbers);
            }
            else if (sortType == 2)
            {
                WriteLine("You have chosen descending order sorting.");
                numbers = Descending(length, numbers);
            }
            else
            {
                WriteLine("Invalid choice");
            }
            Write("Sorted list: ");
            for (i = 0; i < length; i++)
            {
                Write(" " + numbers[i]);
            }
            WriteLine();
        }
    }
}